kafkajobs/logger.go

```diff
@@ -34,10 +34,15 @@ func (l *logger) Level() kgo.LogLevel {
 }
 
 func (l *logger) Log(level kgo.LogLevel, msg string, keyvals ...any) {
-	zf := make([]zap.Field, 0, len(keyvals))
+	var zf []zapcore.Field
 
-	for i := range keyvals {
-		zf = append(zf, zap.Any("kgo_driver", keyvals[i]))
+	if zfTmp, ok := tryFetchKeyValPairs(keyvals); ok {
+		zf = zfTmp
+	} else {
+		zf = make([]zap.Field, 0, len(keyvals))
+		for i := range keyvals {
+			zf = append(zf, zap.Any("kgo_driver", keyvals[i]))
+		}
```

Comment:
I guess that you don't need to do anything here, but just remove `kgo_driver` part completely to have a clean logs:
```go
	for i := range keyvals {
		zf = append(zf, zap.Any("kgo_driver", keyvals[i]))
	}
```

Reply:
Do you think we can be completely sure that `keyvals` always have pairs? 
Like `{"key1", val1, "key2": val2, ...}`

Reply:
You may double-check that, but I think that you should always have an even number of fields. If not, you may always add some placeholder at the beginning/end of the sequence and apply the same algorithm for an even number of vals.
You may also look at the kgo/logger.go for the BasicLogger and implement a writer interface for our logger and pass it to the kgo BasicLogger.